pkg/chains/storage/oci/simple.go

```diff
@@ -0,0 +1,96 @@
+// Copyright 2023 The Tekton Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package oci
+
+import (
+	"context"
+	"encoding/base64"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/pkg/errors"
+	"github.com/sigstore/cosign/v2/pkg/oci/mutate"
+	ociremote "github.com/sigstore/cosign/v2/pkg/oci/remote"
+	"github.com/sigstore/cosign/v2/pkg/oci/static"
+	"github.com/tektoncd/chains/pkg/chains/formats/simple"
+	"github.com/tektoncd/chains/pkg/chains/storage/api"
+	"github.com/tektoncd/chains/pkg/config"
+	"knative.dev/pkg/logging"
+)
+
+// SimpleStorer stores SimpleSigning payloads in OCI registries.
+type SimpleStorer struct {
+	// repo configures the repo where data should be stored.
+	// If empty, the repo is inferred from the Artifact.
+	repo *name.Repository
+	// remoteOpts are additional remote options (i.e. auth) to use for client operations.
+	remoteOpts []remote.Option
+}
+
+var (
+	_ api.Storer[name.Digest, simple.SimpleContainerImage] = &SimpleStorer{}
+)
+
+func NewSimpleStorerFromConfig(cfg config.Config) (*SimpleStorer, error) {
+	var repo *name.Repository
```

Comment:
Same comment about handling `cfg.Storage.OCI.Insecure` here. Probably ok to duplicate given the other implementation is deprecated and will be removed.

Reply:
Yeah long term I want to see if we can refactor this + attestation storer to use https://pkg.go.dev/github.com/google/go-containerregistry@v0.15.2/pkg/v1/remote#Pusher so we can consolidate these.

I think we need to make some upstream changes though (since options like Insecure are name options, not remote options) so I didn't try to tackle this in this PR.